cpp/src/utilities/logger.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/logger.hpp>
+#include <cudf/utilities/export.hpp>
+
+#include <rapids_logger/logger.hpp>
+
+namespace CUDF_EXPORT cudf {
+
+rapids_logger::sink_ptr default_logger_sink()
+{
+  auto* filename = std::getenv("CUDF_DEBUG_LOG_FILE");
+  return (filename == nullptr)
+           ? static_cast<rapids_logger::sink_ptr>(std::make_shared<rapids_logger::stderr_sink_mt>())
```

Comment:
Can we lift the `static_cast` outside of the ternary operator here?

Reply:
The static cast is necessary because both arms of the ternary must return the same type without any automatic casting baked in. This can be fixed by using two separate return statements though, so I'll do that.